### PR TITLE
React and Redux implementation

### PR DESCRIPTION
**Pull request by ulisesedu7**
In this project, I've implemented React&Redux to manage the state of my bookstore application. 

I've used React Redux hooks like useSelector() and useDispatch() to work with the individual actions that make the application work.

### CODE REVIEW

File: src/components/AddNewBook.js [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
+    <div className="add-book-container">
+      <h2>ADD NEW BOOK</h2>
+      <form id="main-form" onSubmit={handleAddSubmit}>
+        <input name="title" type="text" placeholder="Book Title" />
+        <input name="author" type="text" placeholder="Author" />
+        <button type="submit">ADD BOOK</button>
+        <span id="alert-message" />
+      </form>
+    </div>
+  );
```
Comment (mikethreels) [inline src/components/AddNewBook.js]: - [x] in this component you should update the local state A local state with  title, author every time you update (onChange)

File: src/components/AddNewBook.js [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
+      setTimeout(() => {
+        alertMessage.textContent = '';
+      }, 4500);
+    }
+  };
+
+  return (
+    <div className="add-book-container">
+      <h2>ADD NEW BOOK</h2>
+      <form id="main-form" onSubmit={handleAddSubmit}>
```
Comment (mikethreels) [inline src/components/AddNewBook.js]: - [x] onSubmit you are running the entire component again you could just run a function instead.

**Review by mikethreels — APPROVED** [review]
Hi Ulises, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ulisesedu7** [comment]
@mikethreels Thank you Mike for the changes, I'll start working on them right away!

**Pull request merged**